master/types.html

```diff
@@ -207,6 +207,477 @@ <h2 id="Precision">Real number precision</h2>
+with the specific requirements and dependencies listed in this section.
+</p>
+
+<div class="annotation svg2-requirement">
+  <table>
+    <tr>
+      <th>SVG 2 Requirement:</th>
+      <td>Improve the DOM.</td>
+    </tr>
+    <tr>
+      <th>Resolution:</th>
+      <td><a href="http://www.w3.org/2011/10/27-svg-irc#T18-35-49">We will generally improve the SVG DOM for SVG 2.</a></td>
+    </tr>
+    <tr>
+      <th>Purpose:</th>
+      <td>Help authors use the SVG DOM by making it less Java-oriented.</td>
+    </tr>
+    <tr>
+      <th>Owner:</th>
+      <td>Cameron (<a href="http://www.w3.org/Graphics/SVG/WG/track/actions/3273">ACTION-3273</a>)</td>
+    </tr>
+    <tr>
+      <th>Note:</th>
+      <td>See <a href="http://www.w3.org/Graphics/SVG/WG/wiki/SVG_2_DOM">SVG 2 DOM</a> Wiki page.</td>
+    </tr>
+  </table>
+</div>
+
+<div class="annotation svg2-requirement">
+  <table>
+    <tr>
+      <th>SVG 2 Requirement:</th>
+      <td>Improve the SVG path DOM APIs.</td>
+    </tr>
+    <tr>
+      <th>Resolution:</th>
+      <td><a href="http://www.w3.org/2011/10/27-svg-irc#T18-23-23">We will improve the SVG path DOM APIs in SVG 2.</a></td>
+    </tr>
+    <tr>
+      <th>Purpose:</th>
+      <td>Clean up SVGPathSegList interface, and possibly share an API with Canvas.</td>
+    </tr>
+    <tr>
+      <th>Owner:</th>
+      <td>Cameron (no action)</td>
+    </tr>
+  </table>
+</div>
+
+
+<h3 id="SVGDOMDependencies">Dependencies for SVG DOM support</h3>
+
+<p class="ready-for-wider-review">The SVG DOM is defined in terms of <a href="https://heycam.github.io/webidl/">Web IDL</a>
+interfaces. All IDL fragments in this specification must be interpreted as
+required for <a href="https://heycam.github.io/webidl/#dfn-conforming-set-of-idl-fragments">conforming IDL fragments</a>,
+as described in the Web IDL specification. [<a href="refs.html#ref-webidl">WebIDL</a>]</p>
+
+<p>The SVG DOM builds upon a number of DOM specifications.  In particular:</p>
+
+<ul>
+  <li>The SVG DOM requires, at a minimum, complete support for
+  <a href="https://www.w3.org/TR/dom/">DOM4</a>;
+  SVG software should implement
+  the <a href="https://dom.spec.whatwg.org/">latest DOM standard</a> 
+  wherever possible
+  [<a href="refs.html#ref-dom">DOM</a>]</li>
+
+  <li>The SVG DOM requires support for relevant aspects of 
+  <a href="https://www.w3.org/TR/uievents/">UI Events</a> and
+  <a href="https://www.w3.org/TR/clipboard-apis/">Clipboard API and events</a>
+  ([<a href="refs.html#ref-uievents">uievents</a>], [<a href="refs.html#ref-clipboard-apis">clipboard-apis</a>]).
+  (For the specific features that are required, see
+  see <a href="interact.html#RelationshipWithUIEVENTS">Relationship with UI Events</a>.)</li>
+
+  <li>The SVG DOM requires, at a minimum, complete
+  support for all interfaces from
+  <a href="https://www.w3.org/TR/DOM-Level-2-Style/Overview.html">DOM Level 2 Style</a>
+  ([<a href="refs.html#ref-dom-level-2-style">dom-level-2-style</a>])
```

Comment:
We should really not be referring to DOM 2 Style or DOM 2 Views which are both ancient specifications. Views is [defined as obsoleted by HTML5](https://www.w3.org/TR/DOM-Level-2-Views/).

Reply:
Yep, that's why I removed DOM Views completely.

Problem with DOM 2 Style is that _some_ of it is still relevant & required (while some is obsolete, never implemented), but it's replacement, CSS OM, is still only WD status & includes some new features which may not be stable. Which is why I went for the approach of requiring the union of the two specs: 

>  <li>The SVG DOM requires, at a minimum, complete
>   support for all interfaces from
>   <a href="https://www.w3.org/TR/DOM-Level-2-Style/Overview.html">DOM Level 2 Style</a>
>   ([<a href="refs.html#ref-dom-level-2-style">dom-level-2-style</a>])
>   that have not been dropped in the
>   <a href="https://drafts.csswg.org/cssom/">CSS Object Model (CSSOM)</a> specification.
>   SVG software should implement the latest version of
>   <a href="https://drafts.csswg.org/cssom/">CSS Object Model (CSSOM)</a>
>   wherever possible
>   ([<a href="refs.html#ref-cssom-1">cssom-1</a>])
>   </li>

Reply:
Fair enough, I agree with your reasoning